lib/timers/promises.js

```diff
@@ -59,17 +59,24 @@ function setTimeout(after, value, options = {}) {
       lazyDOMException('The operation was aborted', 'AbortError'));
   }
   return new Promise((resolve, reject) => {
-    const timeout = new Timeout(resolve, after, args, false, true);
+    let signalHandler = null;
+    function cleanupSignalHandlerThenResolve(arg) {
+      signal.removeEventListener('abort', signalHandler, { once: true });
+      resolve(arg);
+    }
```

Comment:
I would prefer to not allocate a closure here. This adds overhead.

Reply:
I think you have to capture the handler in a closure since you need a reference to it to remove it.